packages/mui-material-next/src/Switch/Switch.types.ts

```diff
@@ -55,16 +56,17 @@ export interface SwitchProps
   value?: unknown;
 }
 
-/**
- *
- * Demos:
- *
- * - [Switch](https://mui.com/material-ui/react-switch/)
- * - [Transfer List](https://mui.com/material-ui/react-transfer-list/)
- *
- * API:
- *
- * - [Switch API](https://mui.com/material-ui/api/switch/)
- * - inherits [IconButton API](https://mui.com/material-ui/api/icon-button/)
- */
-export default function Switch(props: SwitchProps): JSX.Element;
+export interface SwitchTypeMap<
+  AdditionalProps = {},
+  RootComponentType extends React.ElementType = 'span',
+> {
+  props: SwitchOwnProps & AdditionalProps;
+  defaultComponent: RootComponentType;
+}
+
+export type SwitchProps<
+  RootComponentType extends React.ElementType = SwitchTypeMap['defaultComponent'],
+  AdditionalProps = {},
+> = OverrideProps<SwitchTypeMap<AdditionalProps, RootComponentType>, RootComponentType>;
+
+export interface SwitchOwnerState extends SwitchProps {}
```

Comment:
For the properties with default values (`color`, `edge`, `size`), maybe we could use `NonNullable` as well? e.g. https://github.com/mui/material-ui/blob/master/packages/mui-material-next/src/Chip/Chip.types.ts#L100-L101

Could be something else to consider for https://github.com/mui/material-ui/issues/39935

Reply:
@mj12albert Good idea, but I don't think that will work, because NonNullable removes undefined, but doesn't make the property required if it was optional before. Some TS madness😅. I think the best thing would be to wait until #39939 is merged. Then both, this one and https://github.com/mui/material-ui/pull/39894#discussion_r1400185059 would be fixed.